Replace metricsmap-bpf-prom-sync with Prometheus Collector pattern (#27370)
cilium-team/cilium: Documentation/community/roadmap.rst

Codebase modularization
~~~~~~~~~~~~~~~~~~~~~~~

> As the project is growing in complexity it is becoming increasingly important
> to be able to divide it into more manageable chunks. To achieve this, we're
> working on modularizing the codebase and going from a tightly coupled design
> (one large initialization and configuration) to a more loosely coupled design
> of mostly self-contained modules. This will make Cilium internals easier to
> comprehend, test and extend.

Reviewer: Avoid "we", use plain, clear language, specific is better than general, future tense is OK here
```suggestion
As the project grows in complexity, it is increasingly important
to be able to divide the project into more manageable chunks. To achieve this, contributors
are modularizing the codebase and going from a tightly coupled design
(one large initialization and configuration) to a more loosely coupled design
of mostly self-contained modules. This will make Cilium internals easier to
comprehend, test, and extend.
```

Author: Not part of this PR.

Contributions in this area are very welcome. To get started, take a look at the
:ref:`guide-to-the-hive` documentation and the issues referenced from
`modularization meta issue <modularization-issue_>`_. If you have any questions

> or ideas please join us on the ``#sig-modularization`` channel on `Cilium

Reviewer: Use plain, clear language (avoid "please")
```suggestion
or ideas, join us on the ``#sig-modularization`` channel on `Cilium
```

Author: Not part of this PR.
